Sources/Kitura/RouterRequest.swift

```diff
@@ -102,13 +93,11 @@ public class RouterRequest {
     /// The currently matched section of the URL.
     public internal(set) var matchedPath = ""
 
-    /// The original URL as a string.
-    public var originalURL: String {
-        return serverRequest.urlString
-    }
+    /// The URL from the request if properly received
+    public internal(set) var url : URL?
 
-    /// The URL.
-    public let url: String
+    /// The URL from the request as URLComponents
+    public internal(set) var urlComponents = URLComponents()
```

Comment:
A compromise may be that we keep `originalURL: String` around, and just change `url: String` to `url: URL` so that's the only breaking change.